### PR TITLE
 function that sums items in a list (list can contain integers and lists of integers)

### PR DESCRIPTION
**Pull request by JanetJanx**
Hi @dennisja, could yo please review the sums_items list and its tests?

### CODE REVIEW

File: sum_item_list.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#function to sum up items in a list
+def sum_items(a):
```
Comment (dennisja) [inline sum_item_list.py]: Hey @Janet-Namutebi You were required to use recursion but it so happens that you just used loops, Please read about recursion and re-implement this. Recursion will help you avoid having nested for conditions. For example if I change the program input to add a list in a list in a list as well i.e
```py
 l = [1, 2, [2, 3, [1, 5]]]
```
You will need to write another if statement and a for .... But with recursion, one solution will handle all the nested levels

**Comment by dennisja** [comment]
Example:- 
Let's write a function that calculates the factorial of a number using a loop 
```python
def calc_factorial(n):
    if n <= 1:
        return 1

    factorial = 1
    for i in range(2, n+1):
        factorial *= i
    return factorial
```
Using recursion
```py
def calc_factorial(n):
    if n <= 1:
        return 1
    return n * calc_factorial(n-1)
```
Recursion is 🆒  🙌

File: sum_item_list.py (hunk 1 above)
Comment (JanetJanx) [reply sum_item_list.py]: Alright, let me look at it tomorrow

**Comment by JanetJanx** [comment]
This is a nice intro sample, let me try it out tomorrow

**Comment by JanetJanx** [comment]
Hi @dennisja , I have re-implemented the sum_items function using recursion and implemented the one for the power function. Could you please look through.

File: power_recursion.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+def power_recursion(a,b):
```
Comment (dennisja) [inline power_recursion.py:1]: remove the redundant elif statements

File: power_recursion.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    if type(a) == int and type(b) == int:
+        if b == 0:
```
Comment (dennisja) [inline power_recursion.py]: Cater for cases where the base is 1 and 0

**Review by dennisja — CHANGES_REQUESTED** [review]

File: sum_item_list.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#function to sum up items in a list
+def sum_items(list_number):
```
Comment (dennisja) [inline sum_item_list.py:2]: Remove redundant nestings

File: sum_item_list.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    sum = 0
+    for item in list_number:
+        if type(item) == int:
+            sum += item
+        elif isinstance(item, list):
+                if type(sum_items(item)) == int:
```
Comment (dennisja) [inline sum_item_list.py]: I wonder what you were trying to do here 🤔

File: sum_item_list.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+                    sum += sum_items(item)
+                else:
```
Comment (dennisja) [inline sum_item_list.py]: redundant

**Review by dennisja — CHANGES_REQUESTED** [review]

**Pull request merged**